docs/principles/jsonapi.md

```diff
@@ -115,7 +115,28 @@ Data objects may declare relationships to other resources — "links with struct
 
 `relationships` is a mapping of "relation name" ⇒ "relation object". That relation object must conform to the structure shown above; it should provide `links.related`, and must provide `data.id` and `data.type`.
 
-Relationships must not provide the actual content of the related resource — it may only link to related resources. Links must also declare a [version]() in the URL to the related resource. This should be the *resolved-version* of the requested resource (not the requested version).
+### <a id="relationships-and-expansion"></a>Relationships and Expansion
+
+Snyk REST APIs support expansion of related resources by enriching relationship [resource identifier objects](https://jsonapi.org/format/#document-resource-identifier-objects) with attributes. Snyk APIs use relationship expansion as an alternative to JSON API [Compound Documents](#rough-compound-documents). This is a Snyk extension to the JSON API specification.
+
+```json
+{
+    "id": "7d1bae82-346a-4f7b-a8cb-37c8f159e415", // unique resource uuid
+    "type": "some-resource", // resource type
+    "attributes": { /* some-resource's attributes */ },
+    "relationships": {
+        "other-resource": {
+            "links": { /* ... */ },
```

Comment:
Thanks for thinking through our cases - looking forward to our meeting on Monday. 

I was curious about `links` here - would it be appropriate to define a relationship to a resource that does not yet have an API in the `relationships` dict?  An example is `rule_result` in our model - which is something we might eventually have an API for, but we won't at GA. Right now, we've put those details in the attributes dict.

Is there a case for a relationship with no link?